src/AggregateFunctions/AggregateFunctionGroupArraySorted.h

```diff
@@ -0,0 +1,390 @@
+                {
+                    std::sort(cur_elems.value.begin(), cur_elems.value.begin() + (max_elems * 2));
+                    cur_elems.value.erase(cur_elems.value.begin() + max_elems, cur_elems.value.begin() + (max_elems * 2));
+                    part_sorted = true;
+                }
+                else if (cur_elems.value.size() > max_elems)
+                {
+                    std::sort(cur_elems.value.begin(), cur_elems.value.end());
+                    cur_elems.value.resize(max_elems, arena);
+                    sorted = true;
+                }
+                else if (cur_elems.value.size() == max_elems)
+                {
+                    if (part_sorted)
+                        sorted = true;
+                    else
+                    {
+                        std::sort(cur_elems.value.begin(), cur_elems.value.end());
+                        sorted = true;
+                    }
+                }
+            }
+        }
+    }
+
+    void merge(AggregateDataPtr __restrict place, ConstAggregateDataPtr rhs, Arena * arena) const override
+    {
+        auto & cur_elems = data(place);
+        auto & rhs_elems = data(rhs);
+
+        if (rhs_elems.value.empty())
+            return;
+        mergeNoSampler(cur_elems, rhs_elems, arena);
+    }
+
+    void ALWAYS_INLINE mergeNoSampler(Data & cur_elems, const Data & rhs_elems, Arena * arena) const
+    {
+        UInt64 new_elems;
+        if (limit_num_elems)
+        {
+            if (cur_elems.value.size() >= max_elems)
+                return;
+            new_elems = std::min(rhs_elems.value.size(), static_cast<size_t>(max_elems) - cur_elems.value.size());
+        }
+        else
+            new_elems = rhs_elems.value.size();
+
+        for (UInt64 i = 0; i < new_elems; ++i)
+            cur_elems.value.push_back(rhs_elems.value[i], arena);
+    }
+
+    void serialize(ConstAggregateDataPtr __restrict place, WriteBuffer & buf, std::optional<size_t> /* version */) const override
+    {
+        writeVarUInt(data(place).value.size(), buf);
+        Arena * arena = new Arena;
+        const char * begin = arena->alignedAlloc(sizeof(Node), alignof(Node));
+        ColumnConst * column = nullptr;
+
+        auto & value = data(place).value;
+        for (size_t row_num = 0; row_num < value.size(); ++row_num)
+        {
+            column->insert(value[row_num]);
+            column->serializeValueIntoArena(row_num, *arena, begin, nullptr);
+            Node * node = Node::allocate(*column, row_num, arena);
+            node->write(buf);
+        }
+
+        if constexpr (Trait::last)
+            writeBinaryLittleEndian(data(place).total_values, buf);
+
+        if constexpr (Trait::sampler == Sampler::RNG)
+        {
+            writeBinaryLittleEndian(data(place).total_values, buf);
+            WriteBufferFromOwnString rng_buf;
+            rng_buf << data(place).rng;
+            writeStringBinary(rng_buf.str(), buf);
+        }
```

Comment:
Ok. Now I think we can make it simpler.
Get `ISerialization` from data_type and use `serializeBinary/deserializeBinary`

https://github.com/ClickHouse/ClickHouse/blob/master/src/DataTypes/Serializations/ISerialization.h#L306-L307